make/autoconf/flags-ldflags.m4

```diff
@@ -128,10 +128,8 @@ AC_DEFUN([FLAGS_SETUP_LDFLAGS_HELPER],
     fi
   fi
 
-  if test "x$ENABLE_REPRODUCIBLE_BUILD" = "xtrue"; then
-    if test "x$TOOLCHAIN_TYPE" = xmicrosoft; then
-      REPRODUCIBLE_LDFLAGS="-experimental:deterministic"
-    fi
+  if test "x$TOOLCHAIN_TYPE" = xmicrosoft; then
+    REPRODUCIBLE_LDFLAGS="-experimental:deterministic"
```

Comment:
For the cflag, we check that the compiler supports it, but for the linker flag you are just setting it without a check. Before this patch, if we got here and ENABLE_REPRODUCIBLE_BUILD was true, it meant that the test had passed for the compiler, from which we could assume it would also work for the linker, but that is no longer the case.

Reply:
Good point.

Reply:
> I made some additional cleanup associated with shmem. I think that sysAssert can (and probably should)
> be replaced with SHMEM_ASSERT. I can fix that as well, if someone from serviceability says that I should do it.

Please do it.
There is no reason to keep 2 identical defines.